cloud_automation/gcf/bt_trigger.go

```diff
@@ -0,0 +1,136 @@
+package btcachegeneration
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"time"
+
+	"google.golang.org/api/dataflow/v1b3"
+	"cloud.google.com/go/bigtable"
+	"cloud.google.com/go/storage"
+)
+
+const (
+	projectID          = "google.com:datcom-store-dev"
+	bigtableInstance   = "prophet-cache"
+	bigtableCluster    = "prophet-cache-c1"
+	createTableRetries = 3
+	columnFamily       = "csv"
+	dataflowTemplate   = "gs://datcom-dataflow-templates/templates/csv_to_bt"
+	bigtableNodes      = 300
+)
+
+// GCSEvent is the payload of a GCS event.
+type GCSEvent struct {
+	Name   string `json:"name"`
+	Bucket string `json:"bucket"`
+}
+
+func ReadFromGCS(ctx context.Context, bucketName, fileName string) ([]byte, error) {
+	gcsClient, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Printf("Failed to create gcsClient: %v\n", err)
+		return nil, fmt.Errorf("Failed to create gcsClient: %v", err)
+	}
+
+	bucket := gcsClient.Bucket(bucketName)
+	rc, err := bucket.Object(fileName).NewReader(ctx)
+	if err != nil {
+		log.Printf("Unable to open file from bucket %q, file %q: %v\n", bucketName, fileName, err)
+		return nil, fmt.Errorf("Unable to open file from bucket %q, file %q: %v", bucketName, fileName, err)
+	}
+	defer rc.Close()
+	return ioutil.ReadAll(rc)
+}
+
+func SetupBigtable(ctx context.Context, tableID string) error {
```

Comment:
nit: Add function comments to say it creates table, cluster and scales up nodes?